src/Transport/Receiving/MessageReceiverNotifier.cs

```diff
@@ -137,6 +111,39 @@ public void Start()
             isRunning = false;
         }
 
+        async Task InitializeAndStartReceiver(int i)
+        {
+            try
+            {
+                var internalReceiver = await clientEntities.Get(fullPath, entity.Namespace.Alias)
+                    .ConfigureAwait(false);
+
+                if (internalReceiver == null)
+                {
+                    throw new Exception($"MessageReceiverNotifier did not get a MessageReceiver instance for entity path {fullPath}, this is probably due to a misconfiguration of the topology");
+                }
+
+                var options = new OnMessageOptions
+                {
+                    AutoComplete = false,
+                    AutoRenewTimeout = settings.AutoRenewTimeout,
+                    MaxConcurrentCalls = maxConcurrentCalls
+                };
+                options.ExceptionReceived += OptionsOnExceptionReceived;
+                internalReceivers[i] = internalReceiver;
+                onMessageOptions[i] = options;
+
+                internalReceiver.OnMessage(m => ReceiveMessage(internalReceiver, m, i, pipelineInvocationTasks), options);
+
+                isRunning = true;
+            }
+            catch (Exception ex)
+            {
+                await errorCallback(ex)
```

Comment:
Original code was aggregating exceptions for all failed attempts to create receivers and then throwing that.
Now we invoke an `errorCallback` for each exception thrown to create a reciever.

Reply:
Yeah the aggregation we introduced in the v7 timeframe. I still not sure how valuable it is. But in order to preserve this I've pushed up another commit which aggregates all the exceptions and then calls the critical error API with the aggregated exceptions.